templates/catalog/_partials/subcategories.tpl

```diff
@@ -8,11 +8,18 @@
   {if (isset($display_subcategories) && $display_subcategories eq 1) || !isset($display_subcategories) }
     <div id="subcategories" class="{$componentName} row gx-3">
       {foreach from=$subcategories item=subcategory}
-        <div class="subcategory__wrapper col-6 col-lg-4 col-xl-3"> 
+        <div class="subcategory__wrapper col-6 col-lg-4 col-xl-3">
           <a class="subcategory" href="{$subcategory.url}" title="{$subcategory.name|escape:'html':'UTF-8'}">
             <div class="subcategory__image">
-              {if !empty($subcategory.image.large.url)}
-                <img class="img-fluid" src="{$subcategory.image.large.url}" alt="{$subcategory.name|escape:'html':'UTF-8'}" loading="lazy" width="{$subcategory.image.large.width}" height="{$subcategory.image.large.height}" />
+              {if !empty($subcategory.image.small.url)}
+                <img
+                  class="img-fluid"
+                  src="{$subcategory.image.small.url}"
+                  width="{$subcategory.image.small.width}"
+                  height="{$subcategory.image.small.height}"
+                  alt="{$subcategory.name|escape:'html':'UTF-8'}"
+                  loading="lazy"
+                >
               {/if}
             </div>
```

Comment:
```smarty
            {if !empty($subcategory.image.small.url)}
              <div class="subcategory__image">
                <img
                  class="img-fluid"
                  src="{$subcategory.image.small.url}"
                  width="{$subcategory.image.small.width}"
                  height="{$subcategory.image.small.height}"
                  alt="{$subcategory.name|escape:'html':'UTF-8'}"
                  loading="lazy"
                >
              </div>
            {/if}
```
@Hlavtox no need for any extra loops 😉

But there's another issue. You can't access `$subcategory.image.small` when you remove main category image and use only the miniature, because whole ``$subcategory.image` is null then.

Reply:
This will be fixed for v9, I will do it with you. ;-)